test/utest.cpp

```diff
@@ -80,17 +81,34 @@ private slots:
       QVERIFY(active_states.size() > 0);
 
       // Get the available events for the first active state
-      const QSet<QString> available_events = map.at(active_states.at(0));
+      const std::set<std::pair<QString, QString>> available_events = map.at(active_states.at(0));
       if (available_events.empty())
       {
         std::cout << "State '" << active_states.at(0).toStdString() << "' has no available events" << std::endl;
         break;
       }
 
+      // Get the available events for the first active state
+      std::set<std::pair<QString, QString>> action_ids = sm.getStateTransitionMap().at(active_states.first());
+      if (action_ids.empty())
+      {
+        std::cout << "State '" << active_states.at(0).toStdString() << "' has no available events" << std::endl;
+        break;
+      }
+
+      QList<QString> events;
+      QList<QString>::iterator k;
+      for (auto pair : action_ids)
+      {
+        events.insert(k, pair.first);
+        k += 1;
+      }
```

Comment:
Is the iterator actually necessary here? Seems like a roundabout way of adding something to a list

```suggestion
      QList<QString> events;
      for (auto pair : action_ids)
      {
        events.push_back(pair.first);
      }
```

Reply:
See my comment below about choosing random events